src/map/lua/lua_baseentity.cpp

```diff
@@ -9894,6 +9894,38 @@ void CLuaBaseEntity::recalculateAbilitiesTable()
     PChar->pushPacket(new CCharAbilitiesPacket(PChar));
 }
 
+/************************************************************************
+ *  Function: getPlayersInRange()
+ *  Purpose : Returns a Lua table of players within range of the base entity
+ *  Example : local players = npc:getPlayersInRange(50)
+ *  Notes   : if the passed argument is nil or 0, just returns all players in the zone
+ ************************************************************************/
+
+sol::table CLuaBaseEntity::getPlayersInRange(uint32 dist = 0)
+{
+    auto players = lua.create_table();
+
+    uint32 size = 0;
+    // clang-format off
+    zoneutils::GetZone(m_PBaseEntity->getZone())->ForEachChar([&](CCharEntity* PChar) {
+        if (!dist || distance(PChar->loc.p, m_PBaseEntity->loc.p) < dist)
+        {
+            players.add(CLuaBaseEntity(PChar));
+            size++;
+        }
+    });
+    // clang-format on
+
+    if (!size)
+    {
+        return sol::lua_nil;
```

Comment:
I would still return an empty table here instead of nil, then you can have loops that work without having to add an additional case instead of having to explicitly nil check